swift/apple/FirezoneKit/Sources/FirezoneKit/Stores/AppStore.swift

```diff
@@ -39,6 +40,16 @@ final class AppStore: ObservableObject {
         }
       }
       .store(in: &cancellables)
+
+    tunnel.$status
+      .receive(on: mainQueue)
+      .sink { [weak self] status in
+        guard let self = self else { return }
+        if status == .disconnected && self.shouldSignOutOnDisconnect {
```

Comment:
Just double-checking here -- this will be called upon ConnlibCallback's `onDisconnect` correct? I think `connlib` still expects `onError` *not* to disconnect the tunnel (that function may end up getting removed, now that we're doing the connlib logging in connlib...)

Reply:
This is called if the tunnel disconnects for any reason.

Actually, on second thoughts, we should sign the user out only if connlib is the cause for disconnection. We don't know of any other case, but we should be able to handle the scenario where say the OS stops the tunnel for some reason. In those cases, we should not sign out the user, and allow the user to reconnect (so we should keep the error state icon et al). What do you think?

Reply:
We don't disconnect on `onError`.

Reply:
> we should be able to handle the scenario where say the OS stops the tunnel for some reason. In those cases, we should not sign out the user, and allow the user to reconnect

I worry that unless we have some kind of alert or notification, the user won't know to reconnect the tunnel. Meanwhile their work could be interrupted. In the case where the tunnel disconnects unexpectedly (i.e. not from `onDisconnect`), would it be sensible you think to automatically try to reconnect let's say, 5 times (with a sleep in between)? The outcome of those attempts would be one of three scenarios:

1. Automatic reconnect attempts fail -> Sign user out, log Error to Console.app (can create an issue to improve communication of this error to the user later)
1. Automatic reconnect succeeds -> connlib connects -> `onSetInterfaceConfig` -> tunnel ready
1. Automatic reconnect succeeds -> connlib attempts connection, receives 4xx -> `onDisconnect` -> Sign user out

Reply:
We don’t know in what scenarios this would happen, so I think it’s better we show an alert for now rather than try to reconnect automatically. Let’s see when we encounter this and then figure out if automatically trying to reconnect would be appropriate.

Reply:
@roop That works. Just for my understanding, I'm guessing for example toggling the VPN off in System Settings wouldn't trigger this? Perhaps something weird like bringing down the interface from the shell if `ifconfig utunX down`?

Reply:
Ah, that's helpful to know. Yeah, it might be quite frustrating to the user then automatically reconnect.